### ISSUE TITLE
usermod: user autoware is currently used by process 1

### ISSUE THREAD

**Issue by LiyouZhou** [Autoware-AI/docker#2]
### Required information:

- Operating system and version: Ubuntu
- Autoware installation type: From Source (Docker)
- Autoware version or commit hash 
  - From source: 729cb019
- ROS distribution and version: Melodic
- ROS installation type: in Docker image

### Description of the bug

Getting "usermod: user autoware is currently used by process 1" when running run.sh.

### Steps to reproduce the bug

```sh
$ cd docker/generic
$ ./run.sh -t latest -c off
```

### Expected behavior

Expecting autoware to be launched inside the docker.

### Actual behavior

Getting an error: "usermod: user autoware is currently used by process 1".

**Comment by LiyouZhou** [comment]
I looked into this problem again today.

It is introduced by https://gitlab.com/autowarefoundation/autoware.ai/docker/-/commit/1792f5269a408cbb2acd79301e3ae4b53488cb8e#65a87a8e5ab702ebe98984216236cf145b1aa8f5_136_138

This patch uses `USER autoware` in Dockerfile so that the entrypoint script is now executed in a user shell instead of a root shell. Resulting in the `usermod` command to fail because `usermod` cannot change UID of the user of the current login shell.

In the original implementation, docker container is launch in root shell and the in the entry point, last command `exec gosu autoware "$@"` changes the user to be autoware. 

This is currently broken for anybody who uses ./run.sh to launch the docker container.

People who use `docker run` to launch the container are spared from this problem probably because the --env=USER_ID=$USER_ID option is not specified.

**Comment by LiyouZhou** [comment]
@sgermanserrano we may need to change the solution for https://gitlab.com/autowarefoundation/autoware.ai/docker/-/issues/19

**Comment by LiyouZhou** [comment]
## workaround

use `./run.sh -s` to skip uid check. that mean everything created inside the docker will have uid 1000.

```
    -s,--skip-uid-fix                  Skip uid modification step required when host uid != 1000
```

**Comment by LiyouZhou** [comment]
@sgermanserrano helped me to validate this. because ubuntu by default start to assign uids from 1000, hence if you are the first user on your system, you would have the lucky uid of 1000, which matches the uid of autoware user inside the docker container. Then `usermod` is never run.

However, if your uid happens to not be 1000, then this bug shows up.

**Comment by LiyouZhou** [comment]
found a solution using chroot instead of gosu. patch up soon.

**Comment by JWhitleyWork** [comment]
Closed by #6.
